### PR TITLE
pkg/proc: simplify code with trivial changes

### PR DESCRIPTION
**Pull request by gocurr**

### CODE REVIEW

File: pkg/proc/core/delve_core.go [hunk 1]
```diff
@@ -101,9 +101,6 @@ func threadsFromDelveNotes(p *process, notes []*note) (proc.Thread, error) {
 				return nil
 			}
 			buf := make([]byte, len)
-			if readerr != nil {
```
Comment (aarzilli) [inline pkg/proc/core/delve_core.go:104]: readerr can be set by the call to read.

File: pkg/proc/disasm.go [hunk 2]
```diff
@@ -92,7 +92,7 @@ func firstPCAfterPrologueDisassembly(p Process, fn *Function, sameline bool) (ui
 			continue
 		}
 		if checkPrologue(text, prologue) {
-			r := &text[len(prologue)]
+			r := text[len(prologue)]
```
Comment (aarzilli) [inline pkg/proc/disasm.go:95]: I don't see why I should want to make a copy of the instruction.

Comment (gocurr) [reply pkg/proc/disasm.go]: Thanks. After disassembly,
it is indeed found that some assembly codes will be added if symbol "&" is not used.

**Review by aarzilli — APPROVED** [review]
LGTM

**Pull request merged**